### ISSUE TITLE
intendation of #if code

### ISSUE THREAD

**Issue by stleusc** [uncrustify/uncrustify#1039]
[uncrustify.cfg](http://pastebin.com/kjjH2qwr)

using latest version

Input

```C
void Function(int x)
{
#if (PPDEFINE == TRUE)
    if (x > 3)
    {
#endif
        // do something
        if (x == 0)
        {
            // yes
        }
#if (PPDEFINE == TRUE)
    }
#endif
}

void Function2(int x)
{
    if (x > 3)
    {
        // do something
        if (x == 0)
        {
            // yes
        }
    }
}
```

expected output: same like input!

actual output

```C
void Function(int x)
{
#if (PPDEFINE == TRUE)

    if (x > 3)
    {
#endif

    // do something

    if (x == 0)
    {
        // yes
    }

#if (PPDEFINE == TRUE)
}

#endif
}


void Function2(int x)
{
    if (x > 3)
    {
        // do something

        if (x == 0)
        {
            // yes
        }
    }
}
```

**Comment by CDanU** [comment]
I get somewhat different results on 140d6977cf4cd7e09aa1289b9eb07ac1204930be:

```cpp
void Function(int x)
{
#if (PPDEFINE == TRUE)
    if (x > 3)
    {
#endif
    // do something
    if (x == 0)
    {
        // yes
    }
#if (PPDEFINE == TRUE)
}

#endif
}

void Function2(int x)
{
    if (x > 3)
    {
        // do something
        if (x == 0)
        {
            // yes
        }
    }
}
```

The newline after the 2nd `#if` is falsely caused by `nl_after_func_body              = 2`

Based on the description of `pp_if_indent_code`:
> pp_if_indent_code: Control whether to indent the code between #if, #else and #endif.

I thought this will do what you want, but instead it aligns code __inside__ #if, #else and #endif.

For the provided example this looks like this:
```cpp
void Function(int x)
{
#if (PPDEFINE == TRUE)
        if (x > 3)
        {
#endif
    // do something
    if (x == 0)
    {
        // yes
    }
#if (PPDEFINE == TRUE)
}

#endif
}
```

The `if (x > 3)` is indented in the #if, but the brace in second `#if` is not. The output does not change from uncrustify-0.54 till now.

**Comment by stleusc** [comment]
My basic idea here would be, pretend the #if and #endif are not even there!
Align code at the level they would be without the #if.

I think this is what many people would like it to look like anyway.

**Comment by guy-maurel** [comment]
This is the same problem as #817 
I wrote an explanation on on 21 Nov 2016:
This source shows two control structures:
The first one is a preprocessor structure, the second one a switch/case structure.
The two structures are interweaved.
uncrustify cannot deal with that.

Please put the braces outer the preprocessor statements.
I'll try to create two control structures, but it will be a long work. 
I take it on the TODO list.

**Comment by mihaipopescu** [comment]
Yes. We have actually disabled indentation of preprocessor definitions because of this limitation. I think the feature already available with `pp_ignore_define_body` where actually I think we may have a bug as it takes also `#if` pp blocks.
